### DISCUSSION TITLE
How to get the predict values from many yhats

### DISCUSSION THREAD

**Discussion by saksonita** [ourownstory/neural_prophet#1150] [Q&A - get help using NeuralProphet] [answered]
future = m.make_future_dataframe(data, periods=317)
forecast_future = m.predict(future)
fig = m.plot(forecast_future)

by writing the above code, I've got many yhats (i=365). The plot looks fine, but how can I get those values from the plot? I tried yhat1 but there were many NaN values same as the rest. 

I am looking forward to your response.

**Comment by judussoari** [comment, marked as answer]
Hi @saksonita , thanks for opening a new discussion. 
You probably used auto regression `n_lags>0`. In that case, it's perfectly normal to have `n_lags` NaN values at the beginning of the yhat columns, as no prediction can be made with less than the specified `n_lags` datapoints taken as an input. Your `forecast_future` dataframe contains the predictions you plotted. 
Hope this helps. If not, feel free to post a reproducible code section, so that I can have a closer look. 
Best
Julio

**Reply by saksonita** [reply]
Thank for your fast response. Is there any way to get those predicted values?  Or do I have to loop to get them? 
Thank again and I am looking forward to your response?

**Reply by xeonvs** [reply]
Hi. You may use something like that:
`yhat = prediction[[f'yhat{i}' for i in range(1, predict_period)]].mean(axis=1)`

**Reply by Alan-Wolf-VN** [reply]
If you want to get the latest forecast, you can use something like this: 
latest = m.get_latest_forecast(forecast, include_history_data=True)

It will fetch the latest yhat for each future timestamp
